llama-index-core/llama_index/core/llms/structured_llm.py

```diff
@@ -36,6 +36,22 @@
 )
 
 
+def _escape_json(messages: Sequence[ChatMessage]) -> Sequence[ChatMessage]:
+    """Escape JSON in messages."""
+    new_messages = []
+    for message in messages:
+        if isinstance(message.content, str):
+            new_messages.append(
+                ChatMessage(
+                    role=message.role,
+                    content=message.content.replace("{", "{{").replace("}", "}}"),
```

Comment:
I feel like this could have unintended consequences, like if a user already escaped their input

Reply:
We could also make structured predict not call .format if no kwargs are there

Reply:
It's also notable that template variables have a limited number of characters they can/should use. I have used `r'\{([a-zA-Z0-9-_ ]*?)\}'` in the past.

```
def escape_braces(text):
    pattern = r'\{([a-zA-Z0-9-_ ]*?)\}'
    return re.sub(pattern, r'{{\1}}', text)
```

Reply:
i like that idea, will try this regex. should resolve the "user already escaped their input" issue i think?

Reply:
Ah that's a more complicated regex, involving lookaheads: `(?<!\{)\{([a-zA-Z0-9-_ ]*?)\}(?!\})`

Reply:
Hm thinking about this I might need some help doing regex engineering

There has to be a standardized way to escape JSON right?? is there a library i can just use

Reply:
I was treating this as the inverse problem- making sure that template variables don't look like JSON. Is this an incorrect assumption?